connect/src/com/telenor/connect/ui/ConnectWebViewClient.java

```diff
@@ -87,11 +88,47 @@ public ConnectWebViewClient(
     }
 
     @Override
-    public boolean shouldOverrideUrlLoading(WebView view, String url) {
-        if (ConnectSdk.getRedirectUri() != null
-                && url.startsWith(ConnectSdk.getRedirectUri())) {
-            ConnectUtils.parseAuthCode(url, connectCallback);
-            return true;
+    @SuppressWarnings("unchecked")
+    public boolean shouldOverrideUrlLoading(final WebView view, final String url) {
```

Comment:
I think, the name confusion comes form the fact you are placing code where it does not really belong. See me other comment.